### PR TITLE
Template validation

### PR DESCRIPTION
**Pull request by BNazar00**
Added certificate template validation

### CODE REVIEW

**Review by softservedata — APPROVED** [review]

File: src/main/java/com/softserve/teachua/controller/CertificateTemplateController.java [hunk 1]
```diff
@@ -63,30 +71,69 @@ public List<CertificateTemplatePreview> getAllTemplates() {
...
+     * Use this endpoint to delete template. The controller returns {@code boolean}.
+     * This feature available only for admins.
+     *
+     * @param id - put template id here.
+     * @return {@code boolean}
+     */
+    @AllowedRoles(RoleData.ADMIN)
+    @DeleteMapping("/template/{id}")
+    public boolean deleteTemplate(@PathVariable Integer id) {
+        return certificateTemplateService.deleteTemplateById(id);
```
Comment (Vekeryk) [inline src/main/java/com/softserve/teachua/controller/CertificateTemplateController.java:136]: Do not return boolean. call deleteTemplateById method, then return ResponseEntity.noContent().build()

File: src/main/java/com/softserve/teachua/service/impl/CertificateTemplateServiceImpl.java [hunk 2]
```diff
@@ -90,4 +126,47 @@ public CertificateTemplate getTemplateByFilePath(String filePath) {
...
+                } catch (IOException e) {
+                    log.error("Failed to delete certificate template: error while deleting pdf template " + template);
+                    e.printStackTrace();
+                    return false;
+                }
+                certificateTemplateRepository.deleteById(id);
+                return true;
+            }
+        }
+        return false;
```
Comment (Vekeryk) [inline src/main/java/com/softserve/teachua/service/impl/CertificateTemplateServiceImpl.java:169]: Let this method return void

File: src/main/java/com/softserve/teachua/service/impl/CertificateTemplateServiceImpl.java [hunk 3]
```diff
@@ -90,4 +126,47 @@ public CertificateTemplate getTemplateByFilePath(String filePath) {
...
+                template = templateOptional.get();
+                try {
+                    Path source = Paths.get(
+                        new ClassPathResource("certificates/templates/pdf-templates").getFile().getPath() + "/" +
+                            template.getFilePath());
+                    Files.delete(source);
+                } catch (IOException e) {
+                    log.error("Failed to delete certificate template: error while deleting pdf template " + template);
+                    e.printStackTrace();
+                    return false;
```
Comment (Vekeryk) [inline src/main/java/com/softserve/teachua/service/impl/CertificateTemplateServiceImpl.java:163]: Divide catch to FileNotFoundException and IOException.
You can use custom exception and handle it in CustomExceptionHandler or 
throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Your message") for FileNotFoundException 
and 
throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Your message")
for IOException

You also may use archiveModel(CertificateTemplate certificateType) after deleting, but ArchiveMark methods are not implemented(

File: src/main/java/com/softserve/teachua/service/impl/PdfTemplateServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                os.write(multipartFile.getBytes());
+            }
+            return CertificateTemplateUploadResponse.builder()
+                .fieldsList(certificateByTemplateService.getTemplateFields(file.getPath()))
+                .templateName(templateName)
+                .build();
+
+        } catch (IOException e) {
+            log.error("Error uploading pdf of template");
+            e.printStackTrace();
```
Comment (Vekeryk) [inline src/main/java/com/softserve/teachua/service/impl/PdfTemplateServiceImpl.java:65]: Put e.printStackTrace into log.error

File: src/main/java/com/softserve/teachua/service/impl/PdfTemplateServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        }
+
+        return null;
```
Comment (Vekeryk) [inline src/main/java/com/softserve/teachua/service/impl/PdfTemplateServiceImpl.java:68]: You are returning null after exception, so controller also will return null as string with 200 status code.
Maybe it is possible to clarify IOException with FileNotFoundException (etc.) and throw new ResponseStatusException

File: src/main/java/com/softserve/teachua/service/impl/PdfTemplateServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                .messages(messagesList)
+                .filePath(targetName)
+                .build();
+
+        } catch (IOException e) {
+            log.error("Error saving pdf of template");
+            e.printStackTrace();
+        }
+
+        return null;
```
Comment (Vekeryk) [inline src/main/java/com/softserve/teachua/service/impl/PdfTemplateServiceImpl.java:103]: Same here

**Pull request merged**